typescript/packages/plugins/uniswap/src/tools.ts

```diff
@@ -0,0 +1,49 @@
+import type { DeferredTool, EVMWalletClient } from "@goat-sdk/core";
+import type { z } from "zod";
+import { getApprovalTransaction, getQuote, getSwapTransaction, sendTransaction } from "./api";
+import { CheckApprovalBodySchema, GetQuoteBodySchema, GetSwapBodySchema, SendTransactionBodySchema } from "./types";
+
+export type UniswapToolsOptions = {
+    apiKey: string;
+    baseUrl: string;
+};
+
+export function getTools({ apiKey, baseUrl }: UniswapToolsOptions): DeferredTool<EVMWalletClient>[] {
+    return [
+        {
+            name: "check_approval",
```

Comment:
There is already the approval action on the erc20 plugin. I would leverage that one and remove this one from here. So I can install both erc20 and uniswap plugins and the agent can use both to achieve their goal

Reply:
this is not a generic approval call, it's part of the Uniswap Trading API suite - it does approvals for all sort of functionality including intents and bridging, with and without Permit2. The plugin is not usable without this.